src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/MethodDefHandler.java

```diff
@@ -110,26 +110,15 @@ public void checkIndentation() {
         checkModifiers();
         checkThrows();
 
-        if (getMethodDefParamRightParen(getMainAst()) != null) {
-            checkWrappingIndentation(getMainAst(), getMethodDefParamRightParen(getMainAst()));
-        }
+        final DetailAST nextToLastChild = getMainAst().getLastChild().getPreviousSibling();
```

Comment:
> We really only needed to get the next to last child in any case; whether it was the RPAREN or some child of an IMPLEMENTS_CLAUSE or ....

What if neither exists in AST? Originally code made it clear that `RPAREN` may not exist.

Reply:
I should have provided more details:

```bash
➜  checkstyle git:(master) ✗ cat src/.../indentation/MethodDefHandler.java | grep "Token type:" -A 2 -B 2

        if (getMethodDefParamRightParen(getMainAst()) != null) {
            System.out.println("Token type: " + TokenUtil.getTokenName(getMainAst().getType()));
            checkWrappingIndentation(getMainAst(), getMethodDefParamRightParen(getMainAst()));
        }

➜  checkstyle git:(master) ✗ mvn -Dtest=IndentationCheckTest test | grep "Token type:" | sort | uniq                                        
Token type: ANNOTATION_FIELD_DEF
Token type: CTOR_DEF
Token type: METHOD_DEF
Token type: RECORD_DEF

```
There are a few different AST's that we are calling `MethodDefParamRightParen`, and after noticing that, I thought the present approach was misleading and over-complicated. We only need to get the next to last child in any case, and this is why I changed this.

Reply:
I was asking more on the values of the new code inside the variable `nextToLastChild`. As before, that was always `RPAREN` in the old code, but now your changes expanded it to others. What are the unique token types for that variable. Is the next to last always `RPAREN` and `IMPLEMENTS_CLAUSE` and nothing else?

Reply:
This was a good question:
```bash
➜  checkstyle git:(issue-9152) ✗ cat src/.../indentation/MethodDefHandler.java | grep "Next to last child token type:" -A 2 -B 2

        final DetailAST nextToLastChild = getMainAst().getLastChild().getPreviousSibling();
        System.out.println("Next to last child token type: "
            + TokenUtil.getTokenName(nextToLastChild.getType()));
        checkWrappingIndentation(getMainAst(), nextToLastChild);

➜  checkstyle git:(issue-9152) ✗ mvn -Dtest=IndentationCheckTest test | grep "Next to last child token type:" | sort | uniq                                                  
Next to last child token type: IDENT
Next to last child token type: IMPLEMENTS_CLAUSE
Next to last child token type: LITERAL_DEFAULT
Next to last child token type: LITERAL_THROWS
Next to last child token type: RPAREN

```

Reply:
ok, that is alot more tokens than there was before since we were only doing `RPAREN`.

Can you add new cases for each of those tokens, good and bad. I assume bad would be best if that token was on its own isolated line. Technically your bad can make every identifier/keyword on it's own line to show how everything will be handled.

Example:
````
class
Test
extends
Object
{
}
````

Reply:
https://github.com/checkstyle/checkstyle/issues/9269

@rnveach an unresolvable violation is blocking this test case. How should we proceed?

Reply:
Indentation has quite a few issue areas, I am sure quite a few will fall into this hole. I wouldn't let anything block you if it isn't directly related to your production code changes. You can define them in an issue if you want, but for the most part we can move on.

Reply:
I really can't expand to these test cases too much without running into more unresolvable violations.  I can say that it looks like right now that record definitions are in better shape than class definitions:

```bash

➜  src cat MyClass.java 
interface SimpleInterface {
}

record
SimpleRecord()
    implements
        SimpleInterface {

}

class
SimpleClass
        implements
        SimpleInterface {

}

➜  src /usr/lib/jvm/java-15-jdk/bin/javac --enable-preview --source 15 MyClass.java     
Note: MyClass.java uses preview language features.
Note: Recompile with -Xlint:preview for details.

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.41-SNAPSHOT-all.jar -c config.xml MyClass.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/MyClass.java:5:1: 'SimpleRecord' has incorrect indentation level 0, expected level should be 4. [Indentation]
[ERROR] /home/nick/Desktop/Tester/src/MyClass.java:12:1: 'SimpleClass' has incorrect indentation level 0, expected level should be 4. [Indentation]
Audit done.
Checkstyle ends with 2 errors.

```
If I fix indentation as specified:

```bash

➜  src cat MyClass.java 
interface SimpleInterface {
}

record
    SimpleRecord()
    implements
        SimpleInterface {

}

class
    SimpleClass
    implements
        SimpleInterface {

}

➜  src java -jar ~/IdeaProjects/checkstyle/target/checkstyle-8.41-SNAPSHOT-all.jar -c config.xml MyClass.java
Starting audit...
[ERROR] /home/nick/Desktop/Tester/src/MyClass.java:12:5: 'class def ident' has incorrect indentation level 4, expected level should be 0. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

Reply:
Implementation has changed, this is outdated now.